### PR TITLE
test: added test cases of navigation bar (#487)

### PR DESCRIPTION
**Pull request by sundasnoreen12**
[INF-343](https://2u-internal.atlassian.net/browse/INF-343)
### Description
Added Following cases:
1. Add Navigation Bar Factory
2. API Test Cases
    a. It successfully get navigation tabs
    b. It failed to get navigation tabs
    c. It denied to get navigation tabs
    d. It denied to get navigation bar when user has no access on course
3. CourseTabsNavigation
    a. It should display navigation tabs

#### How Has This Been Tested?
npm run test 

#### Merge Checklist

* [ ] If your update includes visual changes, have they been reviewed by a designer? Send them a link to the Sandbox, if applicable.
* [ ] Is there adequate test coverage for your changes?

#### Post-merge Checklist

* [ ] Deploy the changes to prod after verifying on stage or ask **@openedx/edx-infinity** to do it. 
* [ ] 🎉 🙌 Celebrate! Thanks for your contribution.

### CODE REVIEW

File: src/components/NavigationBar/data/__factories__/navigationBar.factory.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  })
+  .attr('course_access', null, {
+    additional_context_user_message: null,
+    developer_message: null,
+    error_code: null,
+    has_access: true,
+    user_fragment: null,
+    user_message: null,
+  })
+  .sequence('course_id', (idx) => `course-v1-${idx}`)
```
Comment (awais-ansari) [inline src/components/NavigationBar/data/__factories__/navigationBar.factory.js:19]: make it `option` with a default value of `course-v1:edX+DemoX+Demo_Course`.

File: src/components/NavigationBar/data/__factories__/navigationBar.factory.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  .attr('is_enrolled', null, false)
+  .attr('is_self_paced', null, false)
+  .attr('is_staff', null, true)
+  .attr('number', null, 'DemoX')
+  .attr('org', null, 'edX')
+  .attr('original_user_is_staff', null, true)
+  .attr('title', null, 'Demonstration Course')
+  .attr('username', null, 'edx')
+  .option('courseId', null, 'course-v1:edX+DemoX+Demo_Course')
```
Comment (awais-ansari) [inline src/components/NavigationBar/data/__factories__/navigationBar.factory.js:28]: You can use `course_id`. No need to define `courseId` again

File: src/components/NavigationBar/data/thunks.js [hunk 3]
```diff
@@ -14,19 +15,19 @@ export function fetchTab(courseId, rootSlug) {
     dispatch(fetchTabRequest({ courseId }));
     try {
       const courseHomeCourseMetadata = await getCourseHomeCourseMetadata(courseId, rootSlug);
-      if (!courseHomeCourseMetadata.courseAccess.hasAccess) {
```
Comment (awais-ansari) [inline src/components/NavigationBar/data/thunks.js:17]: This condition is required to check the Access permission.

File: src/components/NavigationBar/data/api.test.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+import { initializeStore } from '../../../store';
+import { executeThunk } from '../../../test-utils';
+import { getCourseMetadataApiUrl } from './api';
+import { fetchTab } from './thunks';
+
+import './__factories__';
+
+const courseId = 'course-v1:edX+TestX+Test_Course';
+
```
Comment (awais-ansari) [inline src/components/NavigationBar/data/api.test.js:15]: remove extra line

File: src/components/NavigationBar/data/api.test.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    });
+    axiosMock = new MockAdapter(getAuthenticatedHttpClient());
+    store = initializeStore();
+  });
+
+  afterEach(() => {
+    axiosMock.reset();
+  });
+
+  it('successfully get navigation tabs', async () => {
```
Comment (awais-ansari) [inline src/components/NavigationBar/data/api.test.js:37]: The first character should be capital. Successfully

**Review by awais-ansari — CHANGES_REQUESTED** [review]

File: src/components/NavigationBar/data/__factories__/navigationBar.factory.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+import { getApiBaseUrl } from '../../../../data/constants';
+
+Factory.define('navigationBar')
+  .attr('can_show_upgrade_sock', null, false)
+  .attr('can_view_certificate', null, false)
+  .attr('celebrations', null, {
+    first_section: false, streak_discount_enabled: false, streak_length_to_celebrate: null, weekly_goal: false,
+  })
+  .option('hasCourseAccess', null, true)
+
```
Comment (awais-ansari) [inline src/components/NavigationBar/data/__factories__/navigationBar.factory.js:12]: remove extra line

**Review by awais-ansari — APPROVED** [review]

File: src/components/NavigationBar/data/api.test.js [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+  afterEach(() => {
+    axiosMock.reset();
+  });
+
+  it('Successfully get navigation tabs', async () => {
+    axiosMock.onGet(`${getCourseMetadataApiUrl(courseId)}`).reply(200, (Factory.build('navigationBar', 1)));
+    await executeThunk(fetchTab(courseId, 'outline'), store.dispatch, store.getState);
+
+    expect(store.getState().courseTabs.tabs).toHaveLength(4);
+    expect(store.getState().courseTabs.courseStatus).toEqual('loaded');
```
Comment (ayeshoali) [inline src/components/NavigationBar/data/api.test.js:41]: I believe courseStatus should be checked before tabs

**Review by ayeshoali — APPROVED** [review]
Please rebase before merging

**Pull request merged**